pkg/legacy/converter_test.go

```diff
@@ -143,3 +143,31 @@ func TestBuildConfigProviders(t *testing.T) {
 	assert.Equal(t, "consul", p.Name)
 	assert.Equal(t, "123456", p.Token)
 }
+
+func TestBuildHistogramPercentiles(t *testing.T) {
+	agentConfig := make(Config)
+
+	// empty list
+	agentConfig["histogram_percentiles"] = ""
+	empty := buildHistogramPercentiles(agentConfig)
+	assert.Nil(t, empty)
+
+	// list with invalid values
+	agentConfig["histogram_percentiles"] = "1, 2, -1, 0"
+	var expectedInvalids []string // empty
+	actualInvalids := buildHistogramPercentiles(agentConfig)
+	assert.Equal(t, expectedInvalids, actualInvalids)
```

Comment:
You should use the `Empty` method: https://godoc.org/github.com/stretchr/testify/assert#Empty